pkg/cli/image/manifest/dockercredentials/credentials.go

```diff
@@ -46,13 +48,53 @@ func NewFromFile(path string) (auth.CredentialStore, error) {
 	}, nil
 }
 
+// Basic auth should allow authorizing against Authorization Service whose URL can differ from Registry URL.
+// Documentation: https://docs.docker.com/registry/spec/auth/token/
+// To support this we have to first discover the Authorization Service (realm URL param)
+// and map that to a registry URL (service URL param). We can then use this mapping if Basic lookup fails.
 type keyringCredentialStore struct {
 	credentialprovider.DockerKeyring
 	registryclient.RefreshTokenStore
+	lock                       sync.Mutex
+	authServiceRegistryMapping map[string]string
+}
+
+func (s *keyringCredentialStore) RefreshToken(url *url.URL, service string) string {
```

Comment:
this hacky part (RefreshToken) can be replaced with cleaner approach in https://github.com/openshift/library-go/pull/1205 via AuthorizationServiceRegistryMappingConsumer interface once it merges